assets/js/components/OutfrontTakeoverTool/OutfrontTakeoverTool.tsx

```diff
@@ -71,18 +71,18 @@ class OutfrontTakeoverTool extends React.Component<
       stationScreenOrientationList: {},
     };
 
-    this.toggleAlertWizard = this.toggleAlertWizard.bind(this);
-    this.openModal = this.openModal.bind(this);
-    this.toggleModal = this.toggleModal.bind(this);
-  }
-
-  componentDidMount() {
     fetch(`${BASE_URL}/stations_and_screen_orientations`)
       .then((response) => response.json())
       .then((result) =>
         this.setState({ stationScreenOrientationList: result })
       );
 
+    this.toggleAlertWizard = this.toggleAlertWizard.bind(this);
+    this.openModal = this.openModal.bind(this);
+    this.toggleModal = this.toggleModal.bind(this);
```

Comment:
I totally agree that it is a bit of a mess and needs to be improved, but I wonder if the improvement should be a total conversion to a functional component? I made this [task](https://app.asana.com/0/1185117109217422/1206882196838938/f) last week because I was unhappy with some changes I need to make in two different ways (I implemented a context for some easier state management, but class components can't use them 🙃 ). Do you think it's fine to wait until that task is done?